src/Chapter3.hs

```diff
@@ -910,6 +989,20 @@ Implement instances of "Append" for the following types:
 class Append a where
     append :: a -> a -> a
 
+newtype Gold = Gold Int
+instance Append Gold where
+    append :: Gold -> Gold -> Gold
+    append (Gold a) (Gold b) = Gold $ a + b
+
+newtype List x = List [x]
+instance Append (List x) where
+    append :: List x -> List x -> List x
+    append (List a) (List b) = List $ a ++ b
+
+newtype Challenge x = Challenge (Maybe x)
+instance (Append x) => Append (Challenge x) where
+    append :: Challenge x -> Challenge x -> Challenge x
+    append (Challenge (Just a)) (Challenge (Just b)) = Challenge $ Just $ append a b
```

Comment:
Unfortunately, this function is partial. It doesn't handle cases when one of `Maybe` is `Nothing`

Reply:
Shame on me

I was sure GHC can detect such cases though

Reply:
Yes, GHC detects such cases. Unfortunately, pattern matching exhaustiveness checker is not enabled by default :unamused: It requires special GHC options. We've added them in the `learn4haskell.cabal` file (package metadata file), so if you use our `make` commands, you will see these warnings. But they are not observable in GHCi :(

Reply:
Now that I ran `make build` I see warnings for non-exhaustive matching. 
`make test-chapter3` doesn't show any warnings

Side note: I should explore GHC options before trying to build something bigger than task solutions